optimade/server/entry_collections/entry_collections.py

```diff
@@ -1,34 +1,110 @@
-from abc import abstractmethod
-from typing import Collection, Tuple, List
+from abc import abstractmethod, ABC
+from typing import Tuple, List, Union
+import warnings
+import re
+
+from lark import Transformer
 
-from optimade.server.mappers import BaseResourceMapper
 from optimade.filterparser import LarkParser
 from optimade.models import EntryResource
-from optimade.server.query_params import EntryListingQueryParams
+from optimade.server.config import CONFIG
+from optimade.server.exceptions import BadRequest, Forbidden
+from optimade.server.mappers import BaseResourceMapper
+from optimade.server.query_params import EntryListingQueryParams, SingleEntryQueryParams
+from optimade.server.warnings import FieldNotRecognized
+
 
+class EntryCollection(ABC):
+    """ Backend-agnostic base class for querying collections of
+    [`EntryResource`][optimade.models.EntryResource]s. """
```

Comment:
Again, this will not work until #430 has been merged. So if we just want to merge this PR, this change can wait to be implemented in #430 itself.
```suggestion
    [`EntryResource`][optimade.models.entries.EntryResource]s. """
```